h2o-core/src/main/java/water/parser/ParseSetup.java

```diff
@@ -267,11 +273,16 @@ public static ParseSetup guessSetup( Key[] fkeys, ParseSetup userSetup ) {
     //Calc chunk-size
     // FIXME: should be a parser specific - or at least parser should be able to override defaults
     Iced ice = DKV.getGet(fkeys[0]);
+    
     if (ice instanceof Frame && ((Frame) ice).vec(0) instanceof UploadFileVec) {
       t._gblSetup._chunk_size = FileVec.DFLT_CHUNK_SIZE;
     } else {
-      t._gblSetup._chunk_size = FileVec.calcOptimalChunkSize(t._totalParseSize, t._gblSetup._number_columns, t._maxLineLength,
-              Runtime.getRuntime().availableProcessors(), H2O.getCloudSize(), false /*use new heuristic*/, true);
+      int calculatedChunkSize = FileVec.calcOptimalChunkSize(t._totalParseSize, t._gblSetup._number_columns, t._maxLineLength,
+          Runtime.getRuntime().availableProcessors(), H2O.getCloudSize(), false /*use new heuristic*/, true);
+      int maxExpectedHeaderSize = Math.max(10000, (int)t._maxLineLength * 300); // to make sure we dont lose bits and get a negative
+      int acceptableHeaderSize = Math.min(500000, maxExpectedHeaderSize); //can't expect half a meg of headers
+      int finalChunkSize = Math.max(calculatedChunkSize, acceptableHeaderSize);
```

Comment:
Wow, thanks. The whole chunkSize reevaluation is kind of clumsy there, and in static ParseSetup.guessSetup() there's no way to go ahead and get chunk size individually, per parser, but what we can do is use dataOffset that is already calculated in some parsers (ARFF), and go from there. if the parser does not calculate dataOffset, it's -1, so it has no impact. What do you think? Committing this update now.